conan/tools/premake/premakedeps.py

```diff
@@ -0,0 +1,106 @@
+from conans.util.files import save
+from conans.model.build_info import CppInfo
+from conans.paths import BUILD_INFO_PREMAKE
+
+class PremakeTemplate(object):
+    def __init__(self, deps_cpp_info):
+        self.includedirs = ",\n".join('"%s"' % p.replace("\\", "/")
+                                        for p in deps_cpp_info.includedirs)
+        self.libdirs = ",\n".join('"%s"' % p.replace("\\", "/")
+                                    for p in deps_cpp_info.libdirs)
+        self.bindirs = ",\n".join('"%s"' % p.replace("\\", "/")
+                                    for p in deps_cpp_info.bindirs)
+        self.libs = ", ".join('"%s"' % p.replace('"', '\\"') for p in deps_cpp_info.libs)
+        self.system_libs = ", ".join('"%s"' % p.replace('"', '\\"') for p in deps_cpp_info.system_libs)
+        self.defines = ", ".join('"%s"' % p.replace('"', '\\"') for p in deps_cpp_info.defines)
+        self.cxxflags = ", ".join('"%s"' % p for p in deps_cpp_info.cxxflags)
+        self.cflags = ", ".join('"%s"' % p for p in deps_cpp_info.cflags)
+        self.sharedlinkflags = ", ".join('"%s"' % p.replace('"', '\\"') for p in deps_cpp_info.sharedlinkflags)
+        self.exelinkflags = ", ".join('"%s"' % p.replace('"', '\\"') for p in deps_cpp_info.exelinkflags)
+        self.frameworks = ", ".join('"%s.framework"' % p.replace('"', '\\"') for p in deps_cpp_info.frameworks)
+        self.sysroot = "%s" % deps_cpp_info.sysroot.replace("\\", "/")
+
+class PremakeDeps(object):
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+
+    def generate(self):
+        # Current directory is the generators_folder
+        generator_files = self.content
+        for generator_file, content in generator_files.items():
+            save(generator_file, content)
+
+    def _get_cpp_info(self):
+        ret = CppInfo('','')
```

Comment:
This seems it could be better a ``NewCppInfo()``, that doesn't require args. That is the helper that we use for new integrations to clean things for 2.0